libs/wingsdk/src/target-sim/container.inflight.ts

```diff
@@ -126,53 +138,88 @@ export class Container implements IContainerClient, ISimulatorResourceInstance {
+    this.addTrace(
+      `Starting container from ${this.imageTag}...`,
+      TraceType.RESOURCE
+    );
 
-    // wait until the container is running
-    await waitUntil(async () => {
-      const container = JSON.parse(
-        await runCommand("docker", ["inspect", this.containerName])
-      );
-      return container?.[0]?.State?.Running;
+    this.child = this.dockerSpawn("run", dockerRun, {
+      stdout: true,
+      stderr: true,
+      onError: (err) => {
+        this.addTrace(err.stack ?? err.message, TraceType.LOG, LogLevel.ERROR);
+      },
     });
 
-    if (!this.props.containerPort) {
-      return {};
-    }
-
     let hostPort: number | undefined;
     await waitUntil(async () => {
-      const container = JSON.parse(
-        await runCommand("docker", ["inspect", this.containerName])
-      );
+      if (!this.child?.started) {
+        throw new Error(`container ${this.imageTag} stopped unexpectedly`);
+      }
 
-      hostPort =
-        container?.[0]?.NetworkSettings?.Ports?.[
-          `${this.props.containerPort}/tcp`
-        ]?.[0]?.HostPort;
+      const container = await this.tryInspect(this.containerName);
 
-      return hostPort !== undefined;
+      // if we are waiting for a port, check if the container is listening to it
+      if (this.props.containerPort) {
+        hostPort =
+          container?.[0]?.NetworkSettings?.Ports?.[
+            `${this.props.containerPort}/tcp`
+          ]?.[0]?.HostPort;
+
+        return hostPort !== undefined;
+      }
+
+      // if we are not waiting for a port, just check if the container is running
+      return container?.[0]?.State?.Running;
     });
 
-    if (!hostPort) {
-      throw new Error(
-        `Container does not listen to port ${this.props.containerPort}`
-      );
-    }
+    this.addTrace(
+      `Container ${this.imageTag} started`,
+      TraceType.RESOURCE,
+      LogLevel.INFO
+    );
 
     return {
       [HOST_PORT_ATTR]: hostPort,
     };
   }
 
+  private createEnvFile() {
+    const env = this.props.env ?? {};
+    if (Object.keys(env).length === 0) {
+      return undefined;
+    }
+
+    const envFile = join(Fs.mkdtemp(), "env.json");
+    const envLines = [];
+    for (const k of Object.keys(env)) {
+      envLines.push(`${k}=${env[k]}`);
+    }
+
+    Fs.writeFile(envFile, envLines.join("\n"));
+    return envFile;
+  }
+
+  private async tryInspect(name: string): Promise<any | undefined> {
+    try {
+      return JSON.parse(
+        await this.docker("inspect", [name], {
+          stderr: false,
+          stdout: false,
+        })
+      );
+    } catch {
+      return undefined;
+    }
+  }
+
   public async cleanup(): Promise<void> {
-    this.log(`Stopping container ${this.containerName}`);
-    await runCommand("docker", ["rm", "-f", this.containerName]);
+    this.addTrace(
+      `Stopping container ${this.containerName}`,
+      TraceType.RESOURCE
+    );
+    await this.child?.kill("SIGKILL");
```

Comment:
The given code isn't used

```suggestion
    await this.child?.kill();
```